tools/internal_ci/macos/grpc_master.cfg

```diff
@@ -15,11 +15,16 @@
 # Config file for the internal CI (in protobuf text format)
 
 # Location of the continuous shell script in repository.
-build_file: "grpc/tools/internal_ci/macos/grpc_master.sh"
+build_file: "grpc/tools/internal_ci/macos/grpc_run_tests_matrix.sh"
 timeout_mins: 240
 action {
   define_artifacts {
     regex: "**/*sponge_log.xml"
     regex: "github/grpc/reports/**"
   }
 }
+
+env_vars {
+  key: "RUN_TESTS_FLAGS"
+  value: "-f basictests macos --internal_ci -j 2 --inner_jobs 4"
```

Comment:
Windows workers should have access to bigquery already, but I'm not sure if the google api client python package is installed there.
Also, we'd need to make sure to differentiate between master and pull requests jobs in order to use bq_result_table.
I'll look into that in a followup PR.